build/check-bundle-size.js

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env node
+
+/* eslint-disable */
+
+const jwt = require('jsonwebtoken');
+const github = require('@octokit/rest')();
+const prettyBytes = require('pretty-bytes');
+const fs = require('fs');
+const {execSync} = require('child_process');
+const zlib = require('zlib');
+
+const SIZE_CHECK_APP_ID = 14028;
+const SIZE_CHECK_APP_INSTALLATION_ID = 229425;
+
+const file = process.argv[2];
+const label = process.argv[3];
+const name = `Size - ${label}`;
+
+if (!file || !label) {
+    console.log(`Usage: ${process.argv[0]} ${process.argv[1]} FILE LABEL`);
+    process.exit(1);
+}
+
+const {size} = fs.statSync(file);
+const gzipSize = zlib.gzipSync(fs.readFileSync(file)).length;
+
+process.on('unhandledRejection', error => {
+    console.log(error);
+    process.exit(1)
+});
+
+const pk = process.env['SIZE_CHECK_APP_PRIVATE_KEY'];
+if (!pk) {
+    console.log('Fork PR; not computing size.');
+    process.exit(0);
+}
+
+const key = Buffer.from(pk, 'base64').toString('binary');
+const payload = {
+    exp: Math.floor(Date.now() / 1000) + 60,
+    iat: Math.floor(Date.now() / 1000),
+    iss: SIZE_CHECK_APP_ID
+};
+
+const token = jwt.sign(payload, key, {algorithm: 'RS256'});
+github.authenticate({type: 'app', token});
+
+function getMergeBase() {
```

Comment:
This won't work for the initial push of a branch, when the PR hasn't been opened yet. To support that, see https://github.com/mapbox/mapbox-gl-native/commit/870d7fda4d5b65d9a4993f35322678bc6f50767c.